src/backend/embedding/extract_file/extract_file.py

```diff
@@ -430,9 +440,97 @@ def extract_docx_content(self, customer_guid: str, file_path: str, filename: str
             logger.error(f"Error extracting content from DOCX file '{filename}': {e}")
             raise Exception(f"DOCX content extraction failed: {e}")
 
+    def extract_ppt_content(self, customer_guid: str, file_path: str, filename: str):
+        try:
+
+            results = []
+            presentation = Presentation(file_path)
+
+            for slide_number, slide in enumerate(presentation.slides, start=1):
+                slide_elements = []
+
+                # Extract slide text
+                for shape in slide.shapes:
+                    if shape.has_text_frame:
+                        for paragraph in shape.text_frame.paragraphs:
+                            text = paragraph.text.strip()
+                            if text:
+                                slide_elements.append({
+                                    "type": "text",
+                                    "content": text
+                                })
+
+                    # Extract table data
+                    if shape.has_table:
+                        table = shape.table
+                        table_data = [
+                            [cell.text.strip() for cell in row.cells]
+                            for row in table.rows
+                        ]
+                        table_text = self.format_table_as_text(table_data)
+                        slide_elements.append({
+                            "type": "table",
+                            "content": table_text
+                        })
+
+                    # Extract images
+                    if shape.shape_type == 13:  # 13 indicates a picture
```

Comment:
Make 13 as an ENUM or make sure you get this value from already declared enum in Presentation library